packages/block-library/src/navigation/style.scss

```diff
@@ -686,10 +684,8 @@ button.wp-block-navigation-item__content {
 		font-size: inherit;
 	}
 
-	&:not(.always-shown) {
-		@include break-small {
-			display: none;
-		}
+	:not(.is-collapsed) > & {
```

Comment:
The `>` selector is necessary because the class gets added to the nav element, but it does make the CSS a bit more complex.